icechunk-python/python/icechunk/_icechunk_python.pyi

```diff
@@ -3,6 +3,12 @@ import datetime
 from collections.abc import AsyncGenerator
 from typing import Any
 
+class Ref:
```

Comment:
Is this really the model we want to return to users?

Alternatives could be:
1. Arraylake style Pydantic models: `BranchName(str)` and `TagName(str)`
2. or, 
```
class RefType(Enum):
    BRANCH = auto()
    TAG = auto()

class Ref:
    name: str
    kind: RefType
```

Reply:
Oh sorry, forgot to comment. i like your approach best, but I'd like to see how easy/hard is to pattern match the result in python. Probably tests will show me that.

Other option would be to have separate functions, `list_tags` and `list_branches`, or maybe the three of them.

Reply:
I think that tags and branches can be in separate functions and that simplifies a lot from the Python user perspective